server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -478,4 +485,38 @@ public void clear() {
             runnables.clear();
         }
     }
+
+    private boolean canMatch(LeafReaderContext ctx) throws IOException {
+        boolean canMatch = true;
+        // skip segments for search after if min/max of them doesn't qualify competitive
+        canMatch = canMatch & canMatchSearchAfter(ctx);
+        return canMatch;
```

Comment:
This is effectively just `return canMatchSearchAfter(ctx)`. Can we simplify and remove this indirection?

Reply:
Yes, I modified like that now. 
Initially I was thinking to have some interface CanMatch with method canMatch() and its implementation CanMatchSearchAfter kind of implementation. Range query can too have CanMatchRangeQuery implementation, but later thought simple local methods can do jobs with lesser complexity here.
Bu yeah, lets keep adding such small methods here with ```&&```.